Subject: pppd: Add declarations to eliminate compile warnings
Origin: upstream, commit:50967962addebe15c7a7e63116ff46a0441dc464
From: Paul Mackerras <paulus@samba.org>

This adds declarations for sif6up, sif6down and setifstate in order
to eliminate warnings about implicit function declarations.

Signed-off-by: Paul Mackerras <paulus@samba.org>

diff --git a/pppd/pppd.h b/pppd/pppd.h
index 47e4d9ad40c3..7c341006bc18 100644
--- a/pppd/pppd.h
+++ b/pppd/pppd.h
@@ -657,6 +657,8 @@ int  cifaddr __P((int, u_int32_t, u_int32_t));
 				/* Reset i/f IP addresses */
 #ifdef INET6
 int  ether_to_eui64(eui64_t *p_eui64);	/* convert eth0 hw address to EUI64 */
+int  sif6up __P((int));		/* Configure i/f up for IPv6 */
+int  sif6down __P((int));	/* Configure i/f down for IPv6 */
 int  sif6addr __P((int, eui64_t, eui64_t));
 				/* Configure IPv6 addresses for i/f */
 int  cif6addr __P((int, eui64_t, eui64_t));
diff --git a/pppd/sys-linux.c b/pppd/sys-linux.c
index a0af43b4e368..a40d00c19269 100644
--- a/pppd/sys-linux.c
+++ b/pppd/sys-linux.c
@@ -240,6 +240,7 @@ static void decode_version (char *buf, int *version, int *mod, int *patch);
 static int set_kdebugflag(int level);
 static int ppp_registered(void);
 static int make_ppp_unit(void);
+static int setifstate (int u, int state);
 
 extern u_char	inpacket_buf[];	/* borrowed from main.c */
 
@@ -2321,7 +2322,7 @@ int sif6down (int u)
  * setifstate - Config the interface up or down
  */
 
-int setifstate (int u, int state)
+static int setifstate (int u, int state)
 {
     struct ifreq ifr;
 
-- 
2.1.0

